src/Groups/GAPGroups.jl

```diff
@@ -224,50 +224,64 @@ end
 #
 # - `PermGroup`, `PermGroup`:
 #   The operation is allowed whenever the parents have the same degree,
-#   and the parent is the symmetric group of that degree.
+#   then the parent of the result is the symmetric group of that degree.
 #
 # - `PcGroup`, `PcGroup` and
 #   `FPGroup`, `FPGroup`:
-#   The operation is allowed whenever the parents belong to the same `GapObj`
-#   (in the sense of `==´), and the first of the two groups is taken.
+#   The operation is allowed whenever the parents have the same `GapObj`
+#   (in the sense of `===´),
+#   then the first of the two groups is taken as the parent of the result.
 #
 # - `SubPcGroup`, `SubPcGroup` and
 #   `SubFPGroup`, `SubFPGroup`:
-#   The operation is allowed whenever the parents have the same `full_group`
-#   (in the sense of `==´), and the first of the two groups is taken.
+#   The operation is allowed whenever the `full_group` fields of the parents
+#   have the same `GapObj`
+#   (in the sense of `===´),
+#   then the first of the two groups is taken as the parent of the result.
 #
 # - `SubPcGroup`, `PcGroup` and
 #   `PcGroup`, `SubPcGroup` and
 #   `SubFPGroup`, `FPGroup` and
 #   `FPGroup`, `SubFPGroup`:
 #   The operation is allowed whenever the `full_group` of the `SubPcGroup`
-#   (`SubFPGroup`) is equal to the `PcGroup` (`FPGroup`)
-#   (in the sense of `==´), and the `PcGroup` (`FPGroup`) is taken.
+#   (`SubFPGroup`) and the `PcGroup` (`FPGroup`) have the same `GapObj`
+#   (in the sense of `===´),
+#   then the `full_group` of the `SubPcGroup` (`SubFPGroup`) is taken
+#   as the parent of the result.
+#   Note that we take a group of type `SubPcGroup` (`SubFpGroup`).
```

Comment:
Perhaps we should explain why (i.e. type stability, e.g. when multiplying elements with identical parent)